packages/hawtio/src/plugins/camel/route-diagram/visualization-service.ts

```diff
@@ -85,22 +87,47 @@ class VisualizationService {
     const nodes: CamelNodeData[] = []
     const links: Edge[] = []
     const doc: XMLDocument = parseXML(xml as string)
+
     const allRoutes = doc.getElementsByTagName('route')
 
     for (const route of allRoutes) {
-      const routeId = route.getAttribute('id')
+      const routeId = route.id
       if (!selectedRouteId || !routeId || selectedRouteId === routeId) {
-        this.addRouteXmlChildren(route, nodes, links, '')
+        this.addRouteXmlChildren(route, nodes, links, routeId, '')
       }
     }
-    const camelNodes = nodes.map(node => ({ id: node.id, data: node, position: { x: 0, y: 0 }, type: 'camel' }))
+    //parse stats
+    const camelNodes = nodes.map(node => {
+      return {
```

Comment:
no need for `return`

Reply:
If there is a TS error when trying to return an object, put it between parentheses

https://ultimatecourses.com/blog/return-object-arrow-function